server/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -419,6 +423,8 @@ public void onTimeout(TimeValue timeout) {
             }
         }
 
+        new NodeMetrics(telemetryProvider.getMeterRegistry(), nodeService);
```

Comment:
this feels odd. there is no point creating an instance if we don't plan to use it.
I recon we could refactor this constructor into a static method.

Reply:
Thanks for the comment, at the end instead of having a static method I decided to use make `NodeMetrics` extend `AbstractLifecycleComponent` so that we can start and stop the cache refresh with based on the Node lifecycle.